base/algorithms.jl

```diff
@@ -0,0 +1,119 @@
+module Algorithms
```

Comment:
`find*` methods are defined in `array.jl` even when they apply to any iterator, which isn't ideal but I guess OK.